src/colorize.cr

```diff
@@ -1,3 +1,88 @@
+# The Colorize library allows you to render texts with the given color and text decoration.
+# The `colorize` method is included in classes like `String`, `Int32`, `Array`, `Hash`, etc.
+# As default, it accepts a color argument and changes the foreground color:
+# ```crystal
+# "foo".colorize(:green)
+# 100.colorize(:red)
+# [1,2,3].colorize(:blue)
+# ```
+#
+# Methods for changing foreground color also include:
+# ```crystal
+# "foo".colorize.fore(:green)
+# "foo".colorize.green
+# ```
+#
+# Methods for changing background color include:
+# ```crystal
+# "foo".colorize.back(:green)
+# "foo".colorize.on(:green)
+# "foo".colorize.on_green
+# ```
+#
+# This library also includes text decoration:
+# ```crystal
+# "foo".colorize.mode(:underline)
+# "foo".colorize.underline
+# ```
+# 
+# The `colorize` method returns a `Colorize::Object` instance, 
+# which allows chaining methods together.
+# ```crystal
+# "foo".colorize.fore(:yellow).back(:blue).mode(:underline)
+# ```
+#
+# Convert to `String` for concatenating.
+# ```crystal
+# "foo".colorize(:red).to_s + " bar"
```

Comment:
Bad example, `"#{string} interpolation" is preferred over `string + " concatenation"` and calls `to_s` implicitly.